[Materials] Upstream support for glass materials

pxlcoder opened the pull request:
#### 66fc5e188133f37c2990801c5c665825dbe85243
<pre>
[Materials] Upstream support for glass materials
<a href="https://bugs.webkit.org/show_bug.cgi?id=294306">https://bugs.webkit.org/show_bug.cgi?id=294306</a>
<a href="https://rdar.apple.com/153029937">rdar://153029937</a>

Reviewed by Abrar Rahman Protyasha and Richard Robinson.

Upstream support for the use of `-apple-visual-effect` with glass materials.

Renaming and refactoring to come in follow-up patches.

* Source/WTF/wtf/PlatformHave.h:

`HAVE_MATERIAL_EFFECT` is unused. `HAVE_MATERIAL_HOSTING` is not. Merge the two.

* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AppleVisualEffect.cpp:
(WebCore::CSSPropertyParserHelpers::isKeywordValidForAppleVisualEffect):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/cocoa/WKMaterialHostingSupport.swift:
(MaterialHostingView.resolvedMaterialEffect(for:)):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/AppleVisualEffectTests.mm:
(TestWebKitAPI::TEST(AppleVisualEffect, GlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, GlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsGlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsGlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsSubduedGlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsSubduedGlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, SubduedGlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, SubduedGlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, NoCrashWhenRemovingLayers)):

Canonical link: <a href="https://commits.webkit.org/296094@main">https://commits.webkit.org/296094@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ede99dcaeb6e099f85841ef5ace7b1c9426c7555

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107297 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26979 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17383 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112511 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57831 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109261 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27660 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35479 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81448 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110226 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96708 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61822 "Passed tests") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/106702 "Build is in progress. Recent messages:") | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14838 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57276 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/99886 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91279 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14872 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115612 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/105843 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34363 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25320 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90490 "Passed tests") | 
| [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/106805 "Build is in progress. Recent messages:") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34739 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92958 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90226 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35134 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12917 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30093 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17361 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34285 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39810 "Failed to build and analyze WebKit") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/130160 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34031 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35402 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37386 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35692 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/Platform/cocoa/WKMaterialHostingSupport.swift:
```diff
@@ -84,13 +84,17 @@ private struct MaterialHostingView<P: MaterialHostingProvider>: View {
             return nil
         case .blur:
-            return Material._hostedBlurMaterial
+            return Material._glass(.regular)
```
this is fine for upstreaming, but at some point this should be refactored to be more idiomatic (like omitting the `Material.`

pxlcoder replied:
Noted. Will do when I rename the cases.

pxlcoder committed:
[Materials] Upstream support for glass materials
https://bugs.webkit.org/show_bug.cgi?id=294306
rdar://153029937

Reviewed by Abrar Rahman Protyasha and Richard Robinson.

Upstream support for the use of `-apple-visual-effect` with glass materials.

Renaming and refactoring to come in follow-up patches.

* Source/WTF/wtf/PlatformHave.h:

`HAVE_MATERIAL_EFFECT` is unused. `HAVE_MATERIAL_HOSTING` is not. Merge the two.

* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+AppleVisualEffect.cpp:
(WebCore::CSSPropertyParserHelpers::isKeywordValidForAppleVisualEffect):
* Source/WebKit/Configurations/WebKit.xcconfig:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Platform/cocoa/WKMaterialHostingSupport.swift:
(MaterialHostingView.resolvedMaterialEffect(for:)):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/AppleVisualEffectTests.mm:
(TestWebKitAPI::TEST(AppleVisualEffect, GlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, GlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsGlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsGlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsSubduedGlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, MediaControlsSubduedGlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, SubduedGlassMaterialParsing)):
(TestWebKitAPI::TEST(AppleVisualEffect, SubduedGlassMaterialParsingWithoutUseSystemAppearance)):
(TestWebKitAPI::TEST(AppleVisualEffect, NoCrashWhenRemovingLayers)):

Canonical link: https://commits.webkit.org/296094@main
